### PR TITLE
gx: update blockstore and path

### PR DESCRIPTION
**Pull request by Stebalien**
and apply fixes for changed interfaces

### CODE REVIEW

**Comment by Stebalien** [comment]
I've made this PR so we don't leave these interfaces out-of-sync for too long.

@magik6k would you like to go with this or merge it into https://github.com/ipfs/go-ipfs/pull/5331?

**Comment by Stebalien** [comment]
Blocked on https://github.com/ipfs/go-ipfs-blockstore/pull/10.

**Comment by magik6k** [comment]
You should be able to cherry-pick https://github.com/ipfs/go-ipfs/pull/5331/commits/e22dd351129f32f3e2d97b578ff80b5710e88c7c

**Comment by magik6k** [comment]
I'd do this here, shouldn't take long to merge.

**Comment by Stebalien** [comment]
This is no longer blocked and should be ready for review.

File: filestore/filestore.go [hunk 1]
```diff
@@ -154,6 +154,22 @@ func (f *Filestore) Get(c *cid.Cid) (blocks.Block, error) {
...
+	switch err {
+	default:
+		return -1, err
+	case nil:
+		return size, nil
+	case blockstore.ErrNotFound:
+		// try filestore
+	}
+
+	return f.fm.GetSize(c)
```
Comment (magik6k) [inline filestore/filestore.go]: You should be able to move this into the switch

**Pull request merged**